### PR TITLE
busybox/init: cleanup

### PR DESCRIPTION
**Pull request by antonlacon**
This is a cleanup of `busybox/scripts/init` done as part of my overlay experiment. No feature change is intended here. Commits are broken up into what they are doing, and to improve reviewability.

Big picture:
Remove leading indent: nearly every line has a 2-space indent
Reorder script: Changed script layout to look like

- bring up enough of the system to be able to set all variables
- set variables
- set functions (shared then local)
- bring up the system and hand it off to real init

Style changes: whitespace & line extension changes
Current names: `reset_oe` and `reset_xbmc` triggers to `reset_le` and `reset_kodi`. I'm not aware of any documentation related to these triggers.

This has been booting and updating my rpi for a couple of weeks. I don't have the environment to test all the possible ways busybox/init brings up the system, however.

### CODE REVIEW

File: packages/sysutils/busybox/scripts/init [hunk 1]
```diff
@@ -383,8 +378,7 @@ load_splash() {
     if [ -f "$MODULE_DIR/uvesafb.ko" -a ! -e /dev/fb0 ]; then
       progress "Loading kernel module uvesafb.ko"
       insmod "$MODULE_DIR/uvesafb.ko" && \
-        set_default_res=yes || \
-        progress "... Failed to load kernel module uvesafb, skipping"
+        set_default_res=yes || progress "... Failed to load kernel module uvesafb, skipping"
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/init]: Not sure about this one - make it all one line (` ... && ... || ...`) or keep as 3 lines?

File: packages/sysutils/busybox/scripts/init [hunk 2]
```diff
@@ -470,12 +464,12 @@ force_fsck() {
   read -t120 -n1
   # The exit status is 0 if input is available
   # The exit status is greater than 128 if the timeout is exceeded
-  if [ "$?" -ne "0" -o "$?" -gt "128" ] ; then
+  if [ "$?" -ne "0" -o "$?" -gt "128" ]; then
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/init]: Side issue, but not sure why we are mixing strings with unary comparisons, shouldn't this be:
```
   if [ $? -ne 0 -o $? -gt 128 ]; then
```
as `$?` will only ever be an integer.

File: packages/sysutils/busybox/scripts/init [hunk 3]
```diff
@@ -470,12 +464,12 @@ force_fsck() {
...
     echo "Repairing filesystem..."
     echo ""
     /usr/sbin/fsck -T -M -y $RUN_FSCK_DISKS
     FSCK_RET="$?"
-    if [ "$(( $FSCK_RET & 8 ))" = 8 ] ; then
+    if [ "$(( $FSCK_RET & 8 ))" = 8 ]; then
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/init]: Same as above, shouldn't this be:
```
     if [ $(( $FSCK_RET & 8 )) -eq 8 ]; then
```
?

And again with all the similar `$FSCK_RET & #` examples below.

**Comment by MilhouseVH** [comment]
Thanks for this, splitting it up into different commits helped enormously with the review!

I've pointed out a couple of nits, not anything you've introduced, but something that's bothered me in the past. I'll include this in nightlies.

I personally would keep this back for 10.0, unless there's a pressing need for it in 9.2?

**Comment by MilhouseVH** [comment]
Regarding the final commit, renaming `reset_oe` and `reset_xbmc`, these files are created in https://github.com/LibreELEC/service.libreelec.settings:
```
neil@nm-linux:~/projects/pullrequest_repos/service.libreelec.settings$ git grep reset_xbmc
src/defaults.py:    'XBMC_RESET_FILE': '%s/reset_xbmc' % CONFIG_CACHE,
src/resources/lib/modules/system.py:                            'action': 'reset_xbmc',
src/resources/lib/modules/system.py:    def reset_xbmc(self, listItem=None):
src/resources/lib/modules/system.py:            self.oe.dbg_log('system::reset_xbmc', 'enter_function', 0)
src/resources/lib/modules/system.py:            self.oe.dbg_log('system::reset_xbmc', 'exit_function', 0)
src/resources/lib/modules/system.py:            self.oe.dbg_log('system::reset_xbmc', 'ERROR: (' + repr(e) + ')', 4)

neil@nm-linux:~/projects/pullrequest_repos/service.libreelec.settings$ git grep XBMC_RESET_FILE
src/defaults.py:    'XBMC_RESET_FILE': '%s/reset_xbmc' % CONFIG_CACHE,
src/resources/lib/modules/system.py:    XBMC_RESET_FILE = None
src/resources/lib/modules/system.py:                reset_file = open(self.XBMC_RESET_FILE, 'w')

neil@nm-linux:~/projects/pullrequest_repos/service.libreelec.settings$ git grep reset_oe
src/defaults.py:    'LIBREELEC_RESET_FILE': '%s/reset_oe' % CONFIG_CACHE,
src/resources/lib/modules/system.py:                            'action': 'reset_oe',
src/resources/lib/modules/system.py:    def reset_oe(self, listItem=None):
src/resources/lib/modules/system.py:            self.oe.dbg_log('system::reset_oe', 'enter_function', 0)
src/resources/lib/modules/system.py:            self.oe.dbg_log('system::reset_oe', 'exit_function', 0)
src/resources/lib/modules/system.py:            self.oe.dbg_log('system::reset_oe', 'ERROR: (' + repr(e) + ')', 4)

neil@nm-linux:~/projects/pullrequest_repos/service.libreelec.settings$ git grep LIBREELEC_RESET_FILE
src/defaults.py:    'LIBREELEC_RESET_FILE': '%s/reset_oe' % CONFIG_CACHE,
src/resources/lib/modules/system.py:    LIBREELEC_RESET_FILE = None
src/resources/lib/modules/system.py:                reset_file = open(self.LIBREELEC_RESET_FILE, 'w')
```
so you'd need to make the corresponding changes there too, and sync the change with this PR.

It might be possible for a user to perform an LE or Kodi hard reset with an update file already in the `/storage/.update` folder, in which case the system could upgrade from an image that respects `reset_oe`/`reset_xbmc` to one that does not, so that the hard reset doesn't happen. And then, as the old reset triggers will not have been removed, the user could downgrade back to a `reset_oe`/`reset_xbmc` version at which point the hard reset kicks in, by which time it might not be wanted.

Which is to say, changing the names isn't entirely straight forward. Personally, I'd keep them as they are.

File: packages/sysutils/busybox/scripts/init (hunk 1 above)
Comment (antonlacon) [reply packages/sysutils/busybox/scripts/init]: Reduced to one line.

**Comment by antonlacon** [comment]
The reset_* commit is dropped. fsck* functions have been altered to working against ints instead of strings. Changeset has booted my rpi without incident.

**Comment by MilhouseVH** [comment]
Thanks, will test and include in nightlies, hopefully merge early next week.

**Comment by MilhouseVH** [comment]
Upgraded, booted and rebooted RPi/RPi2/Generic without incident. Let's get a few days testing with other configurations and see if anything crops up.

**Pull request merged**